app/models/manageiq/providers/ibm_cloud/power_virtual_servers/cloud_manager/event_parser.rb

```diff
@@ -0,0 +1,44 @@
+module ManageIQ::Providers::IbmCloud::PowerVirtualServers::CloudManager::EventParser
+  def self.event_to_hash(event, ems_id)
+    event_hash = {
+      :event_type => "#{event[:resource]}.#{event[:action]}",
+      :source     => "IBMCloud-PowerVS",
+      :ems_id     => ems_id,
+      :ems_ref    => event[:eventID],
+      :timestamp  => event[:time],
+      :full_data  => event,
+      :username   => event[:user][:email]
+    }
+
+    _log.debug("event_hash=#{event_hash}")
+    case event_hash[:event_type]
+    when /^pvm-instance/
+      parse_vm_event!(event, event_hash)
+    end
+
+    event_hash
+  end
+
+  def self.parse_vm_event!(event, event_hash)
+    pvm_instance_name = event[:message].split('\'')[1]
+    ems = ExtManagementSystem.find(event_hash[:ems_id])
+    vm = ems.vms.find_by(:name => pvm_instance_name)
+
+    if vm.nil? && ['update', 'create'].include?(event['type'])
+
+      # TODO: If event type is 'pvm-instance.update' and the VM name was
+      #       changed then only the new name is given in the event message :(
+      #       Should we refresh the entire EMS on this event? Same question
+      #       for a 'pvm-instance.create'.
+      event_hash
```

Comment:
I'm still trying to figure out how to queue up a full EMS refresh here.

Reply:
You could handle this in your `EventTargetParser`